lib/logic/child_rewards_cubit.dart

```diff
@@ -24,42 +23,45 @@ class ChildRewardsCubit extends ReloadableCubit {
+	  UIChild child = _activeUser();
+		ObjectId caregiverID = child.connections.first;
 
 		List<Reward> rewards = [];
-		if(caregiver != null) 
-			rewards = await _dataRepository.getRewards(caregiverId: caregiver.id);
+		if(caregiverID != null) 
+			rewards = await _dataRepository.getRewards(caregiverId: caregiverID);
 
-		List<ChildReward> claimedRewards = await _dataRepository.getChildRewards(childId: child.id);
 		Map<ObjectId, int> claimedCount = Map<ObjectId, int>();
-		claimedRewards.forEach((element) => claimedCount[element.id] = !claimedCount.containsKey(element.id) ? 1 : claimedCount[element.id] + 1);
+		child.rewards.forEach((element) => claimedCount[element.id] = !claimedCount.containsKey(element.id) ? 1 : claimedCount[element.id] + 1);
 
 	  emit(ChildRewardsLoadSuccess(
-			rewards.map((reward) => UIReward.fromDBModel(reward)).where((reward) => reward.limit != null ? reward.limit.compareTo(claimedCount[reward.id] ?? 0) > 0: true).toList(),
-			claimedRewards.map((reward) => UIChildReward.fromDBModel(reward)).toList(),
-			(child as UIChild).points
+			rewards.map((reward) => UIReward.fromDBModel(reward)).where((reward) => reward.limit != null ? reward.limit < (claimedCount[reward.id] ?? 0) : true).toList(),
+			child.rewards,
+			child.points
 		));
   }
 
 	void claimReward(UIReward reward) async {
     UIChild child = _activeUser();
-		Map<CurrencyType, int> points = child.points;
+		List<UIPoints> points = child.points;
+		List<UIChildReward> rewards = child.rewards;
 		ChildReward model = ChildReward(
 			id: reward.id,
 			name: reward.name, 
 			cost: Points.fromUIPoints(reward.cost),
 			icon: reward.icon,
 			date: TimeDate.now()
 		);
-
-		if(points[reward.cost.type] >= reward.cost.quantity) {
-			points[reward.cost.type] -= reward.cost.quantity;
-			await _dataRepository.updateUser(child.id, points: points.entries.map((e) =>
-				Points.fromUICurrency(UICurrency(type: e.key, title: child.pointsNames[e.key]), e.value)).toList()
-			);
-			_authBloc.add(AuthenticationActiveUserUpdated(child.copyWith(points: points)));
-			await _dataRepository.createChildReward(child.id, model).then((value) => doLoadData());
+		UIPoints pointCurrency = points.firstWhere((element) => element.type == reward.cost.type, orElse: () => null);
+		
+		if(pointCurrency != null && pointCurrency.quantity >= reward.cost.quantity) {
+			emit(DataLoadInitial());
+			points[points.indexOf(pointCurrency)] = pointCurrency.copyWith(quantity: pointCurrency.quantity - reward.cost.quantity);
+			await _dataRepository.claimChildReward(child.id, reward: model, points: points.map((e) =>
+				Points.fromUICurrency(UICurrency(type: e.type, title: e.title), e.quantity, creator: e.createdBy)).toList()
+			).then((value) {
+				_authBloc.add(AuthenticationActiveUserUpdated(child.copyWith(points: points, rewards: rewards..add(UIChildReward.fromDBModel(model)))));
+				doLoadData();
```

Comment:
You are trying to reload? no need for `DataLoadInitial` and `doLoadData()` - just emit new `ChildRewardsLoadSuccess` with updated fields

Reply:
The reward limit logic is there. I need to know if the reward should be displayed or not. That's why there is `doLoadData()`. And without `DataLoadInitial `rewards page isn't refresh (`ChildRewardsLoadSuccess `from `doLoadData()` don't do anything after initial, first call).

Reply:
Emitting the same state should refresh as long as there is any change in the fields (and those fields ale listed in props)

Reply:
Regarding the logic you are only updating a single reward so in the case when all the last available one was claimed it needs to be removed from the list (otherwise no change, right?). And you already have all rewards in your state so no need to query again.